wgpu-hal/src/gles/command.rs

```diff
@@ -603,10 +621,25 @@ impl crate::CommandEncoder<super::Api> for super::CommandEncoder {
         &mut self,
         _layout: &super::PipelineLayout,
         _stages: wgt::ShaderStages,
-        _offset: u32,
-        _data: &[u32],
+        start_offset: u32,
+        data: &[u32],
     ) {
-        unimplemented!()
+        let range = self.cmd_buffer.add_push_constant_data(data);
+
+        let end = start_offset + data.len() as u32 * 4;
+        let mut offset = start_offset;
+        while offset < end {
+            let uniform = self.state.push_offset_to_uniform[offset as usize % 4].clone();
+            let size = uniform.size;
+            if uniform.location.is_none() {
+                panic!("No uniform for push constant");
+            }
+            self.cmd_buffer.commands.push(C::SetPushConstants {
+                uniform,
+                range: (range.start + offset)..range.end,
```

Comment:
should this end with `range.start + offset + size` instead of `range.end`?

Reply:
Yes, but since the range only specifies where in `data_bytes` is our push constant data, the end functions more like a bound check, so if we leave the end as the end of the allocation it serves the same purpose.

Technically we only need the start offset

Reply:
let's remove the end entirely then, no need to carry it around for validation